### PR TITLE
[BEAM-121] Additional APIs for registering DisplayData

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @bjchambers 

This PR is based on feeback from #123, #124, #125, #126. I will update those PRs to consume these APIs once merged.

File: sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java [hunk 1]
```diff
@@ -101,6 +125,10 @@ public String toString() {
     return builder.toString();
   }
 
+  private static String convertNamespace(Class<?> nsClass) {
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/transforms/display/DisplayData.java]: `namespaceOf(...)`? `classToNamespace`? (just thinking out loud, feel free to ignore if you don't think they're improvements).

**Comment by bjchambers** [comment]
LGTM

---

Only thing that occurred to me is that we could reduce the number of methods on the builder by putting more logic into the `Type` enum. Let's stick with this for now since the other PRs are based on the existing API, but it may be worth thinking about something like:

`Type.STRING.entry("key", "value")`
`Type.STRING.entryIfNotNull("key", value)`
`Type.STRING.entryIfNotDefault("key", value, defaultValue)`

Then the builder API just needs a method to add an entry if its not null:

`builder.add(Type.STRING.entryIfNotDefault("key", value, "default"))`

This is also nice when including the inference:

`builder.add(Type.infer(value).entry("key", value))`

**Comment by swegner** [comment]
Backported via https://github.com/GoogleCloudPlatform/DataflowJavaSDK/pull/216

**Pull request closed without merging**